app/admin/applications/new/page.tsx

```diff
@@ -55,42 +63,121 @@ export default function NewApplicationPage() {
           />
         </div>
         <div>
-          <label htmlFor="clientId" className="block mb-1">
-            Client ID
+          <label htmlFor="appType" className="block mb-1">
+            Application Type
           </label>
-          <input
-            type="text"
-            id="clientId"
-            name="clientId"
-            className="w-full p-2 border rounded text-black"
-          />
-        </div>
-        <div>
-          <label htmlFor="clientSecret" className="block mb-1">
-            Client Secret
-          </label>
-          <input
-            type="password"
-            id="clientSecret"
-            name="clientSecret"
-            className="w-full p-2 border rounded text-black"
-          />
-        </div>
-        <div>
-          <label htmlFor="apiEndpoint" className="block mb-1">
-            API Endpoint
-          </label>
-          <input
-            type="url"
-            id="apiEndpoint"
-            name="apiEndpoint"
-            placeholder="https://api.example.com"
+          <select
+            id="appType"
+            name="appType"
+            required
+            value={selectedAppType}
+            onChange={(e) => setSelectedAppType(e.target.value)}
             className="w-full p-2 border rounded text-black"
-          />
+          >
+            <option value="">Select an application type...</option>
+            {AppTypes.map((type) => (
+              <option key={type} value={type} className="capitalize">
+                {type.toLowerCase().replace(/_/g, ' ')}
+              </option>
+            ))}
+          </select>
         </div>
+
+        {selectedAppType === "TWITTER" && (
```

Comment:
We should use a `switch case` syntax, as we are about to add several app types in the future, and that way we could use a `default` for the last one. each case should have its own component